slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -83,19 +83,20 @@ trait MySQLProfile extends JdbcProfile { profile =>
       val BitPattern = "^b'([01])'".r
       override def default = meta.columnDef.map((_,tpe)).collect{
         case (v,"String")    => Some(Some(v))
-        case ( BooleanPattern(v) , "Boolean") => Some( Some(if(v eq "1") true else false))
-        case ( BitPattern(v) , "Boolean") => Some( Some(if(v eq "1") true else false))
+        case ( BooleanPattern(v) , "Boolean") => Some( Some(v == "1"))
+        case ( BitPattern(v) , "Boolean") => Some( Some( v == "1"))
       }.getOrElse{
         val d = super.default
         if(meta.nullable == Some(true) && d == None){
           Some(None)
         } else d
       }
-      override def length: Option[Int] = {
-        val l = super.length
-        if(tpe == "String" && varying && l == Some(65535)) None
-        else l
+      override def length: Option[Int] = super.length match{
+          case Some(l) if varying && l >= 65535 => None
+          case l => l
```

Comment:
not a big deal. but the idiomatic way to do this would be `super.length.filter(!varying || _ < 64435)`